test/intersectionWith.js

```diff
@@ -10,4 +10,16 @@ describe('intersectionWith', function() {
     eq(R.intersectionWith(eqA, Ro, So), [{a: 3}, {a: 4}]);
   });
 
+  it('only returns elements from the first list', function() {
```

Comment:
Okay. Shall we deprecate `intersectionWith` (and leave its behaviour as it is now) and define `innerJoin`?